java/org/apache/tomcat/util/IntrospectionUtils.java

```diff
@@ -136,6 +138,19 @@ public static boolean setProperty(Object o, String name, String value,
                         if (actualMethod != null) {
                             actualMethod.append(method.getName()).append("(InetAddress.getByName(\"").append(value).append("\"))");
                         }
+                        // Try a setFoo ( Path )
+                    } else if ("java.nio.file.Path".equals(paramType
+                            .getName())) {
+                        try {
+                            params[0] = Paths.get(value);
+                        } catch (InvalidPathException ipe) {
+                            if (log.isDebugEnabled())
+                                log.debug("IntrospectionUtils: Unable to resolve path:" + value);
```

Comment:
Use the common `[..]` pattern

Reply:
Not following, [..] ?

Reply:
All log statements in Tomcat guard placeholders with `[]`, e.g., `"Did not find file [foo.xml]"`.

Reply:
I've fixed this one.